lib/pull-lib.test.ts

```diff
@@ -0,0 +1,367 @@
+import fs from "fs";
```

Comment:
This file was renamed from `pull.test.ts` to `pull-lib.test.ts` -- most of code shown as changed I didn't actually touch. I will leave a comment on the things that were changed.